### DISCUSSION TITLE
Sleep mode

### DISCUSSION THREAD

**Discussion by fa1321tx** [jgyates/genmon#1111] [Ideas]
First off love the Genmon system! Big thanks to everyone involved in making it what it is today. I was wondering if there is a Sleep Mode or if it's worth considering as a new feature? 
The idea being during the middle of the night I do not really need my generator to kick on waking me up if the power is out for a coupe of hours. I would prefer to sleep through it and my food isn't going to spoil in that window.

**Comment by jgyates** [comment]
Hi @fa1321tx 

No there is not, mainly because the air cooled generac models do not support switching to "Off" mode via modbus. The liquid cooled models do support this but this is not a current feature of genmon. The industrial models supported probably would not want this functionality.

**Reply by fa1321tx** [reply]
Could this be done by adding some sort of relay and timer. I guess that might be outside of the scope of Genmon's abilities?

**Reply by jgyates** [reply]
I don't know how that would happen. There are two wires that controller uses to signal the transfer switch, one is a control signal and the other powers a relay in the transfer switch. Inserting another control relay would not stop the generator from starting, it would only prevent the transfer switch from engaging. Other than that I don't know how to keep the generator from starting other than switch main switch to the Off position.

I also will leave this question for some of the others that my have more knowlege than I do in this area.

**Reply by fa1321tx** [reply]
Thanks!! I know it could be powered off with a electrical contact. Is there a way to send a low voltage signal durning a set time frame that would be easy to toggle on/off in the Genmon. I live in an area where the power going off in the middle of the night for an hour or two is becoming pretty frequent so this would be a worthy endeavor for me.

**Reply by jgyates** [reply]
Are you proposing to wire a relay to the off switch on the controller? or some type of mechanism that pushes the button on the outside of the controller?

The raspberry pi can provide a low voltage output via a GPIO signal. Modifying the add on gengpio.py could be used to add a timer to activate and deactivate the GPIO output. 

Let me know how you intend to implement this. I can potentially add this functionality to the add on if you can provide details on the implementation that could be incorporated in the project wiki (where are you making the relay connections, a wiring diagram, recommended part number, etc). This would allow others to benefit from this modification also.

**Reply by fa1321tx** [reply]
I was thinking of something simple that would just prevent it from starting. For instance on the circuit board J2-5 & J2-7 complete the crank and run circuit. If you placed a normally closed relay that would hold that circuit open during the Sleep Mode it would not start. I image it would generate some alarms in the system when activated.   

[Schematic 1 LG MASTER[1].pdf](https://github.com/jgyates/genmon/files/15192261/Schematic.1.LG.MASTER.1.pdf)

**Reply by jgyates** [reply]
sounds reasonable. I will add this functionality to the feature list. I will update this thread when this is complete.

**Reply by jgyates** [reply]
Maybe worth a test first to make sure it does not trigger an alarm as @gm58 suggested. An alarm preventing the system from starting would defeat the purpose you intend.

**Reply by gm58** [reply]
post disappeared

Wouldn't the example given throw an Alarm that would require going to the controller to reset? If so, wouldn't that defeat the purpose of doing this remotely? I guess it might save you one trip to the generator.

**Comment by fa1321tx** [comment]
> Maybe worth a test first to make sure it does not trigger an alarm as @gm58 suggested. An alarm preventing the system from starting would defeat the purpose you intend.
I agree it needs a test run. The weather has been horrible lately so I will test it soon as these storms back off.

**Reply by jgyates** [reply]
I went ahead and added this functionality in the GPIO output add on (gengpio.py). This add on is configured with the config file i/etc/genmon/gengpio.conf. The updated source, which you can get from updating the softare from the about page in the web interface, also updates the comments in the ./genmon/conf/gengpio.conf file:

https://github.com/jgyates/genmon/blob/3314c6226b62dffc0993d227faab5aa13ece85f8/conf/gengpio.conf#L53

I would do the following:

1) update your genmon via the about page.
2) copy the new gengpio.conf file to /etc/genmon
 
    cd genmon
     sudo cp ./genmon/conf /etc/genmon

3) edit the newly copied file /etc/genmon/gengpio.conf to update these entries:

PIN_EXTERNAL_SIGNAL
PIN_EXTERNAL_SIGNAL_ON_TIME
PIN_EXTERNAL_SIGNAL_ONF_TIME

You will likely need to disable on GPIO pin in the config file by setting it to zero. This will free up the GPIO pin to be used for PIN_EXTERNAL_SIGNAL. 

Let me know how your test goes and if you have any issues.
